src/main/java/org/candlepin/service/exception/subscription/SubscriptionUnknownActivationException.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.exception.subscription;
+
+
+/**
+ * The SubscriptionUnknownActivationException is used to reject an activation attempt for a
+ *  * subscription.
+ */
+public class SubscriptionUnknownActivationException extends SubscriptionServiceException {
```

Comment:
This looks like another potential candidate to be replaced by a suggestion in an earlier comment, to have a generic `ServiceRuntimeException` instead. Then, in the SubscriptionResource block:

```java
           else if (e instanceof SubscriptionUnknownActivationException) {
                message = i18n.tr(
                    "The system is unable to redeem the requested subscription: {0}", dellAssetTag);
            }
```

Instead of catching `SubscriptionUnknownActivationException`. it can catch the `ServiceRuntimeException` and still provide the same error message.